cluster-autoscaler/utils.go

```diff
@@ -187,6 +187,32 @@ func resetPodScheduledConditionForPod(kubeClient *kube_client.Client, pod *kube_
 	return fmt.Errorf("Expected condition PodScheduled")
 }
 
+// FilterOutSchedulable return list of pods which are unschedulable.
+func FilterOutSchedulable(unschedulableCandidates []*kube_api.Pod, nodes []*kube_api.Node, allPods []*kube_api.Pod, predicateChecker *simulator.PredicateChecker) []*kube_api.Pod {
+	unschedulablePods := []*kube_api.Pod{}
+	nodeNameToNodeInfo := createNodeNameToInfoMap(allPods, nodes)
+
+	for _, pod := range unschedulableCandidates {
+		if nodeName, err := predicateChecker.FitsAny(pod, nodeNameToNodeInfo); err == nil {
+			glog.Warningf("Pod %s marked as unschedulable can be scheduled on %s. Ignoring in scale up.", pod.Name, nodeName)
+		} else {
+			unschedulablePods = append(unschedulablePods, pod)
+		}
+	}
+
+	return unschedulablePods
+}
+
+func createNodeNameToInfoMap(pods []*kube_api.Pod, nodes []*kube_api.Node) map[string]*schedulercache.NodeInfo {
```

Comment:
Please add a TODO to refactor the function in scheduler utils so that we don't need this one.